src/glyph_item_iter.rs

```diff
@@ -0,0 +1,229 @@
+    pub struct GlyphItemIter(Boxed<pango_sys::PangoGlyphItemIter>);
+
+    match fn {
+        copy => |ptr| pango_sys::pango_glyph_item_iter_copy(mut_override(ptr)),
+        free => |ptr| pango_sys::pango_glyph_item_iter_free(ptr),
+        init => |_ptr| (),
+        clear => |_ptr| (),
+        get_type => || pango_sys::pango_glyph_item_iter_get_type(),
+    }
+}
+
+impl GlyphItemIter {
+    pub fn init_end(glyph_item: &GlyphItem, text: &str) -> Option<GlyphItemIter> {
+        unsafe {
+            let mut iter = GlyphItemIter::uninitialized();
+            let ret = from_glib(pango_sys::pango_glyph_item_iter_init_end(
+                iter.to_glib_none_mut().0,
+                mut_override(glyph_item.to_glib_none().0),
+                text.to_glib_none().0,
+            ));
+            if ret {
+                Some(iter)
+            } else {
+                None
+            }
+        }
+    }
+
+    pub fn init_start(glyph_item: &GlyphItem, text: &str) -> Option<GlyphItemIter> {
+        unsafe {
+            let mut iter = GlyphItemIter::uninitialized();
+            let ret = from_glib(pango_sys::pango_glyph_item_iter_init_start(
+                iter.to_glib_none_mut().0,
+                mut_override(glyph_item.to_glib_none().0),
+                text.to_glib_none().0,
+            ));
+            if ret {
+                Some(iter)
+            } else {
+                None
+            }
+        }
+    }
+
+    pub fn next_cluster(&mut self) -> bool {
+        unsafe {
+            from_glib(pango_sys::pango_glyph_item_iter_next_cluster(
+                self.to_glib_none_mut().0,
+            ))
+        }
+    }
+
+    pub fn prev_cluster(&mut self) -> bool {
+        unsafe {
+            from_glib(pango_sys::pango_glyph_item_iter_prev_cluster(
+                self.to_glib_none_mut().0,
+            ))
+        }
+    }
+
+    pub fn start_index(&self) -> i32 {
+        self.0.start_index
+    }
+
+    pub fn start_char(&self) -> i32 {
+        self.0.start_char
+    }
+
+    pub fn start_glyph(&self) -> i32 {
+        self.0.start_glyph
+    }
+
+    pub fn end_index(&self) -> i32 {
+        self.0.end_index
+    }
+
+    pub fn end_char(&self) -> i32 {
+        self.0.end_char
+    }
+
+    pub fn end_glyph(&self) -> i32 {
+        self.0.end_glyph
+    }
+
+    pub fn into_data(&self) -> GlyphItemIteratorData {
+        GlyphItemIteratorData {
+            start_glyph: self.0.start_glyph,
+            end_glyph: self.0.end_glyph,
+            start_index: self.0.start_index as usize,
+            end_index: self.0.end_index as usize,
+            start_char: self.0.start_char as usize,
+            end_char: self.0.end_char as usize,
+        }
+    }
+}
+
+pub struct GlyphItemIteratorData {
+    pub start_glyph: i32,
+    pub start_index: usize,
+    pub start_char: usize,
+
+    pub end_glyph: i32,
+    pub end_index: usize,
+    pub end_char: usize,
+}
+
+pub struct GlyphItemIterator<'a> {

```

Comment:
For extra fun you can make the `is_reverse` bool a type parameter for the struct and then get rid of the `if` in `next()` and let it all be decided at compile-time.

Reply:
I will try that too

Reply:
Hm, for this I need write two more iterator impls ?

Reply:
Yes but each would be half as big. No if/else